cli/src/tty.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    ffi::OsString,
+    fmt::Write,
+    io::{Error, ErrorKind, Read},
+    time::Duration,
+};
+
+use benchmark_common::{deserialize, serialize, IncomingMessage, OutgoingMessage, SuiteStatus};
+use serialport::TTYPort;
+
+type Line = Result<String, Error>;
+
+/// SerialConnection, representing a connection over a serial TTYPort
+pub struct SerialConnection {
+    port: TTYPort,
+}
+
+impl SerialConnection {
+    /// Creates a new SerialConnection connected to the provided tty
+    ///
+    /// # Arguments
+    ///
+    /// * `tty` - the path to the tty to connect to
+    pub fn new(tty: &OsString) -> Result<SerialConnection, serialport::Error> {
+        let term = SerialConnection {
+            port: serialport::new(tty.to_string_lossy(), 9600)
+                .timeout(Duration::from_secs(60))
+                .open_native()?,
+        };
+
+        Ok(term)
+    }
+
+    /// Reads a single line form the serial port
+    pub fn read_line(&mut self) -> Line {
+        let mut buf = [0];
+        let mut msg = String::new();
+
+        loop {
+            if self.port.read(&mut buf)? > 0 {
+                let c = buf[0] as char;
+                if c == '\n' {
+                    return Ok(msg);
+                } else {
+                    msg.push(buf[0] as char)
+                }
+            } else {
+                return Err(Error::from(ErrorKind::UnexpectedEof));
+            }
+        }
+    }
+}
+
+impl Write for SerialConnection {
+    fn write_str(&mut self, s: &str) -> std::fmt::Result {
```

Comment:
You may get a similar performance boost with a bufwriter, but not sure in this case as the message is written line by line.